datashader/reductions.py

```diff
@@ -1159,6 +1223,75 @@ def _finalize(bases, **kwargs):
         raise NotImplementedError("mode is currently implemented only for rasters")
 
 
+class where(FloatingReduction):
+    def __init__(self, selector: Reduction, lookup_column: str):
```

Comment:
Needs a docstring with a good example.

Reply:
I have added a docstring. It is somewhat tortuous English when written in a generic form so I have added a concrete example that should help.

I am using the argument name `selector` because long-term I would like to divide the `Reduction` class hierarchy into two:

- `Selection` reductions that use values from a column without modifying them, e.g. `first` and `max`.
- `Combination` reductions that do some form of mathematical combination of the values from a column, e.g. `mean`, `std`, `count`.

Given that division, a `where` reduction will accept a `Selection` but not a `Combination` to select values from the `lookup_column`.

Reply:
Ok, sounds good. Agreed that it's tortuous, but the example does help a lot.